rules/repositories.bzl

```diff
@@ -58,20 +58,52 @@ def _get_bazel_version():
     return struct(major = 0, minor = 0, patch = 0)
 
 def rules_ios_dependencies(
```

Comment:
Changes here:

- Update versions to minimum required for bzlmod
- Move into one additional branch for bzlmod dependencies (i.e. things that support bzlmod)
  - With bzlmod these are loaded in `MODULE.bazel` so they need to be filtered here but with older WORKSPACE we default to include these